### PR TITLE
FOUR-18612: Populate cases_started table with existing requests

### PR DESCRIPTION
**Pull request by rodriquelca**
## Solution
-  add populate cases_started table during migration
## How to Test
Describe how to test that this solution works.
run `php artisan upgrade` command
## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-18612

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.
c:next

### CODE REVIEW

File: upgrades/2024_10_09_032151_populate_case_started.php [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+    /**
+     * Run the upgrade migration.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $caseNumber = 1;
+        $chunkSize = self::CHUNK_SIZE;
```
Comment (devmiguelangel) [inline upgrades/2024_10_09_032151_populate_case_started.php:44]: Remove this line
```suggestion
```

File: upgrades/2024_10_09_032151_populate_case_started.php [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        echo ' ', $count, PHP_EOL;
+
+        // Truncate the table cases_started
+        DB::table('cases_started')->truncate();
+        echo PHP_EOL;
+        echo '    Populating case_started from process_requests';
+
+        try {
+            $this->getNonSystemRequests()
+                ->orderBy('process_requests.id')
```
Comment (devmiguelangel) [inline upgrades/2024_10_09_032151_populate_case_started.php:57]: Remove this line
```suggestion
```

File: upgrades/2024_10_09_032151_populate_case_started.php [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+                'process_requests.status',
+                'process_requests.initiated_at',
+                'process_requests.created_at',
+                'process_requests.completed_at',
+                'process_requests.updated_at',
+                'process_requests.name',
+                'process_requests.parent_request_id',
+                'processes.id as process_id',
+                'processes.name as process_name',
+            );
```
Comment (devmiguelangel) [inline upgrades/2024_10_09_032151_populate_case_started.php:133]: Put the order by clause here
```suggestion
            )
            ->orderBy('process_requests.id');
```

File: upgrades/2024_10_09_032151_populate_case_started.php [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+
+        // Truncate the table cases_started
+        DB::table('cases_started')->truncate();
+        echo PHP_EOL;
+        echo '    Populating case_started from process_requests';
+
+        try {
+            $this->getNonSystemRequests()
+                ->orderBy('process_requests.id')
+                ->chunk($chunkSize, function ($requests) use (&$caseNumber, $startTime, $count) {
```
Comment (devmiguelangel) [inline upgrades/2024_10_09_032151_populate_case_started.php:58]: Use the `CHUNK_SIZE` constant directly
```suggestion
                ->chunk(self::CHUNK_SIZE, function ($requests) use (&$caseNumber, $startTime, $count) {
```

File: database/migrations/2024_09_09_181717_create_cases_started_table.php [hunk 5]
```diff
@@ -14,7 +13,7 @@ public function up(): void
         Schema::create('cases_started', function (Blueprint $table) {
             $table->id();
             $table->unsignedInteger('case_number')->unique();
-            $table->unsignedInteger('user_id');
+            $table->unsignedInteger('user_id')->nullable();
```
Comment (devmiguelangel) [inline database/migrations/2024_09_09_181717_create_cases_started_table.php:16]: Why the `user_id` was changed to `null`?
```suggestion
            $table->unsignedInteger('user_id')->nullable();
```

Comment (rodriquelca) [reply database/migrations/2024_09_09_181717_create_cases_started_table.php:16]: because  user_id is null for  some process like (Expense Report)

**Comment by boliviacoca** [comment]
In sync with @rodriquelca this is no longer applicable, so it is being closed.

**Pull request closed without merging**